src/client/user/components/CreateUrlModal/CreateLinkForm.tsx

```diff
@@ -28,57 +26,34 @@ import { CollapsibleMessageType } from '../../../app/components/CollapsibleMessa
 import FileInputField from '../../widgets/FileInputField'
 import userActions from '../../actions'
 import { GAEvent } from '../../../app/util/ga'
+import { GoGovReduxState } from '../../../app/reducers/types'
+import FormStartAdorment, { TEXT_FIELD_HEIGHT } from './FormStartAdorment'
 
-// Height of the text field in the create link dialog.
-const TEXT_FIELD_HEIGHT = 44
-
-const FormStartAdorment = ({ children }) => {
-  const classes = useCreateLinkFormStyles({
-    textFieldHeight: TEXT_FIELD_HEIGHT,
-  })
-  return (
-    <InputAdornment className={classes.startAdorment} position="start">
-      <Typography className={classes.startAdormentText} color="textSecondary">
-        {children}
-      </Typography>
-    </InputAdornment>
-  )
+type CreateLinkFormProps = {
+  onSubmitLink: () => {},
+  onSubmitFile: (file: File | null) => {},
 }
 
-const mapStateToProps = (state) => ({
-  shortUrl: state.user.shortUrl,
-  longUrl: state.user.longUrl,
-  isUploading: state.user.isUploading,
-  createShortLinkError: state.user.createShortLinkError,
-  uploadFileError: state.user.uploadFileError,
-})
-
-const mapDispatchToProps = (dispatch) => ({
-  setShortUrl: (shortUrl) => dispatch(userActions.setShortUrl(shortUrl)),
-  setLongUrl: (longUrl) => dispatch(userActions.setLongUrl(longUrl)),
-  setRandomShortUrl: () => dispatch(userActions.setRandomShortUrl()),
-  setUploadFileError: (error) =>
-    dispatch(userActions.setUploadFileError(error)),
-  setCreateShortLinkError: (error) =>
-    dispatch(userActions.setCreateShortLinkError(error)),
-})
-
-function CreateLinkForm({
+const CreateLinkForm: FunctionComponent<CreateLinkFormProps> = ({
   onSubmitLink,
-  shortUrl,
-  setShortUrl,
-  longUrl,
-  setLongUrl,
-  setRandomShortUrl,
-  isUploading,
   onSubmitFile,
-  uploadFileError,
-  setUploadFileError,
-  createShortLinkError,
-  setCreateShortLinkError,
-}) {
+}) => {
+  const shortUrl = useSelector((state: GoGovReduxState) => state.user.shortUrl)
+  const longUrl = useSelector((state: GoGovReduxState) => state.user.longUrl)
+  const isUploading = useSelector((state: GoGovReduxState) => state.user.isUploading)
+  const createShortLinkError = useSelector((state: GoGovReduxState) => state.user.createShortLinkError)
+  const uploadFileError = useSelector((state: GoGovReduxState) => state.user.uploadFileError)
+
+  const dispatch = useDispatch()
+  const setShortUrl = (shortUrl: string) => dispatch(userActions.setShortUrl(shortUrl))
+  const setLongUrl = (longUrl: string) => dispatch(userActions.setLongUrl(longUrl))
+  const setRandomShortUrl = () => dispatch(userActions.setRandomShortUrl())
+  const setUploadFileError = (error: string | null) => dispatch(userActions.setUploadFileError(error))
+  const setCreateShortLinkError = (error: string | null) => dispatch(userActions.setCreateShortLinkError(error))
```

Comment:
can I check what happens when null is passed to the `error` argument? is there a default message, or are we needlessly calling this dispatch?

Reply:
The 'null' is use to clear any existing messages from the error message when there is change in file upload / submit link button is clicked. 
It is required to allow the user to press the submit link again and clear previous error message.

However, it seems like we do not need an extra type 'null' but an empty string will do. I will make necessary changes to remove the redundant type